docs/source/notes/autograd.rst

```diff
@@ -3,119 +3,398 @@
 Autograd mechanics
 ==================
 
-This note will present an overview of how autograd works and records the
-operations. It's not strictly necessary to understand all this, but we recommend
-getting familiar with it, as it will help you write more efficient, cleaner
-programs, and can aid you in debugging.
+This note will present an overview of how autograd works. It's not strictly necessary
+to understand all this, but we recommend getting familiar with it, as it will help you
+write more efficient, cleaner programs, and can aid you in debugging.
 
-.. _excluding-subgraphs:
+TL-DR:
 
-Excluding subgraphs from backward
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+- Mathematical formulation: we compute vector-Jacobian products using the chain rule. We support inplace operations using hidden versioning of the Tensors.
+- Non-gradients: we provide helper functions to be able to compute a result that is not the gradient of the user's whole program.
+- Will inplace operation work?: Because they modify the original value, if anything needs this value for the backward, the backward pass will fail.
+- Gradients of 0 vs independent of the input: We don't see the difference.
+- Gradients of 0: They can be represented by a Tensor full of ``0`` s, ``None`` in python, undefined tensor in c++ or and error for ``autograd.grad(allow_unused=False)``.
+- Non-differentiable functions: We do not provide anyone guarantee, but we have fixed rules on what each elementary operation's backward pass should compute which should cover most cases reasonably.
+- Implementation details: Only if you want to understand the current implementation at a high level.
+
+
+
+Mathematical formulation
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+What does the autograd package do
+"""""""""""""""""""""""""""""""""
+
+The autograd package allows to use automatic differentiation techniques on top of PyTorch Tensors.
+Automatic differentiation allows to compute dot product between a given vector and the Jacobian of a user-defined function.
+In this document, we will consider the function :math:`f: x \rightarrow y` where :math:`x` is a vector of size :math:`I` and :math:`y` is a vector of size :math:`O`.
+For simplicity, in this document, we will always consider inputs and ouputs to be 1D, in practice, they can be Tensors with an arbitrary number of dimensions.
+The jacobian matrix associated with this function is that matrix :math:`J_f` of size :math:`O \times I` such that each entry is
```

Comment:
Jacobian (capitalize)

Reply:
Honestly, for everyone who is not a physicist (and that's most DL practitioners), the explanation of automatic differentiation in terms of Jacobians obscures more than it helps.